src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/SchemaGenerator.cs

```diff
@@ -392,7 +395,16 @@ private OpenApiSchema CreateObjectSchema(DataContract dataContract, SchemaReposi
                     ? GenerateSchemaForMember(dataProperty.MemberType, schemaRepository, dataProperty.MemberInfo, dataProperty)
                     : GenerateSchemaForType(dataProperty.MemberType, schemaRepository);
 
-                if ((dataProperty.IsRequired || customAttributes.OfType<RequiredAttribute>().Any())
+                var markNonNullableTypeAsRequired = (dataProperty.MemberInfo?.IsNonNullableReferenceType() ?? false)
+                    && _generatorOptions.MarkNonNullableReferenceTypesAsRequired
+#if !NETSTANDARD2_0
+                    && !_mvcOptions.Value.SuppressImplicitRequiredAttributeForNonNullableReferenceTypes
+#endif
+                    ;
+
+                if ((dataProperty.IsRequired
+                     || customAttributes.OfType<RequiredAttribute>().Any()
+                     || markNonNullableTypeAsRequired)
```

Comment:
It would be cheaper in terms of performance to swap these so that if `markNonNullableTypeAsRequired` no reflection query to find `[Required]` is needed.